spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/functions/SparkFunctions.java

```diff
@@ -52,6 +55,24 @@ public static List<String> list() {
 
   public static UnboundFunction load(String name) {
     // function resolution is case-insensitive to match the existing Spark behavior for functions
-    return FUNCTIONS.get(name.toLowerCase(Locale.ROOT));
+    String lowerCaseName = name.toLowerCase(Locale.ROOT);
+    UnboundFunction function = FUNCTIONS.get(lowerCaseName);
+
+    SparkSession sparkSession = SparkSession.active();
+    String enabledStr =
+        sparkSession
+            .conf()
+            .get(
+                SparkSQLProperties.SYSTEM_FUNC_PUSH_DOWN_ENABLED,
+                String.valueOf(SparkSQLProperties.SYSTEM_FUNC_PUSH_DOWN_ENABLED_DEFAULT));
+    if (Boolean.parseBoolean(enabledStr)
+        && SparkV2Filters.SUPPORTED_FUNCTIONS.contains(lowerCaseName)) {
+      // Use a UnboundFunctionWrapper to wrap the actual system function. The system function will
+      // be
+      // analyzed into ApplyFunctionExpression which could be pushed down datasource.
+      return new UnboundFunctionWrapper(function);
```

Comment:
As I said above, having a boolean that controls whether we wrap to trick Spark into not using `StaticInvoke` is a problem. I think that we probably need to support constant folding with `StaticInvoke`, and should probably also fix this on the Spark side. I don't see why we need to do this in Iceberg.